lib/portage/manifest.py

```diff
@@ -463,16 +463,17 @@ def validateSignature(self):
 
     def addFile(self, ftype, fname, hashdict=None, ignoreMissing=False):
         """Add entry to Manifest optionally using hashdict to avoid recalculation of hashes"""
-        if ftype == "AUX" and not fname.startswith("files/"):
-            fname = os.path.join("files", fname)
-        if not os.path.exists(self.pkgdir + fname) and not ignoreMissing:
+        if ftype == "AUX":
+            if not fname.startswith("files/"):
+                fname = os.path.join("files", fname)
+            if fname.startswith("files"):
+                fname = fname[6:]
```

Comment:
I'm commenting this just in case Matt or someone else has an idea: is there some general way of doing the "starts-with-prefix-if-it-does-trim-it-off" pattern?

It looks like not really, according to https://stackoverflow.com/questions/13236875/python-replace-with-if-line-starts-with-keyword. Oh well! Not a comment against your change, just something I've been wondering for ages in my own stuff.